### PR TITLE
10/06/24 testing

### PR DESCRIPTION
**Pull request by daniel-sussman**
Modified CategorySidebar so that it counts jobs in each category according to the filters applied to *other* categories only. This makes it possible to see select more than one filter in the same category. Otherwise they all disappear once a filter is selected. For example, there are 5 jobs in London and 4 jobs in Reading that were posted in the past 3 days (I've selected 'Last 3 Days' in the #posted category). When I select 'Reading', the 5 London jobs should vanish from the main part of the page but the category_sidebar should still show me that I have the option of selecting London jobs as well. The job count next to 'London' should remain '5', indicating that I will see 5 additional jobs if I select this box.

The modification is to avoid this (no options visible other than what I've selected):
Screenshot 2024-06-11 at 14 29 28

And also meaningless counts appearing within a radio-button category, like this:
Screenshot 2024-06-11 at 14 30 50

Added tests:

- category_sidebar_spec
- job_filter_spec

### CODE REVIEW

**Comment by patanj101** [comment]
Interesting ... I have been working on the same problematic last weekend on a page name "/opportunities" .
I am curious to have your opinion about my solution on the same issue.

https://github.com/Ches-ctrl/Cheddar/commit/edbc5ec31af802fa3c919eaebdd317de526d17bb

File: config/application.rb
Comment (Ches-ctrl) [inline config/application.rb:1]: Do we need this autoload path? Doesn't rails handle this automatically?

Comment (patanj101) [reply config/application.rb:1]: Nice catch. We don't need it. I have created similar folders for Queries & Tasks objects without needed it

**Review by Ches-ctrl — CHANGES_REQUESTED** [review]
Can you remove the autoload?

**Comment by daniel-sussman** [comment]
Charlie, thanks for making those changes! I've sorted the FactoryBot problem, so this should be good to go.

**Comment by daniel-sussman** [comment]
> Interesting ... I have been working on the same problematic last weekend on a page name "/opportunities" . I am curious to have your opinion about my solution on the same issue.
> 
> [edbc5ec](https://github.com/Ches-ctrl/Cheddar/commit/edbc5ec31af802fa3c919eaebdd317de526d17bb)

Thanks for sharing this, JB! I need to take some time to understand what you're doing with facets. However, I'm interested in this line:

```
  def facet_oppotunities(exemption)
    FacetOpportunitiesFetcher.call(Job.all, @params.except(exemption))
  end
```
It seems like this will make several database calls, since the method's called by every category. Is it possible to eager_load Job.all so that we only need to access the database once? Or have you done that in a way that I've overlooked? I had this question when making my own edits to CategorySidebar but couldn't find an elegant solution.

**Review by Ches-ctrl — APPROVED** [review]
Approve

**Pull request merged**